src/libraries/System.Private.CoreLib/src/System/IndexOfAnyAsciiSearcher.cs

```diff
@@ -0,0 +1,345 @@
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe bool TryLastIndexOfAny<TNegator>(ref short searchSpace, int searchSpaceLength, ReadOnlySpan<char> asciiValues, out int index)
+            where TNegator : struct, INegator
+        {
+            Debug.Assert(searchSpace >= Vector128<short>.Count);
+
+            if (IsSupported)
+            {
+                Vector128<byte> bitmap = default;
+                if (TryComputeBitmap(asciiValues, (byte*)&bitmap, out bool needleContainsZero))
+                {
+                    index = Ssse3.IsSupported && needleContainsZero
+                        ? LastIndexOfAnyVectorized<TNegator, Ssse3HandleZeroInNeedle>(ref searchSpace, searchSpaceLength, bitmap)
+                        : LastIndexOfAnyVectorized<TNegator, Default>(ref searchSpace, searchSpaceLength, bitmap);
+                    return true;
+                }
+            }
+
+            index = default;
+            return false;
+        }
+
+        private static int IndexOfAnyVectorized<TNegator, TOptimizations>(ref short searchSpace, int searchSpaceLength, Vector128<byte> bitmap)
+            where TNegator : struct, INegator
+            where TOptimizations : struct, IOptimizations
+        {
+            ref short currentSearchSpace = ref searchSpace;
+
+            if (searchSpaceLength > 2 * Vector128<short>.Count)
+            {
+                // Process the input in chunks of 16 characters. If the input length is a multiple of 16, don't consume
+                // the last 16 characters in this loop. Let the fallback below handle it instead. This is why the condition is
+                // ">" instead of ">=" above, and why "IsAddressLessThan" is used instead of "!IsAddressGreaterThan".
+                ref short twoVectorsAwayFromEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength - (2 * Vector128<short>.Count));
+
+                do
+                {
+                    Vector128<short> source0 = Vector128.LoadUnsafe(ref currentSearchSpace);
+                    Vector128<short> source1 = Vector128.LoadUnsafe(ref currentSearchSpace, (nuint)Vector128<short>.Count);
+
+                    Vector128<byte> result = IndexOfAnyLookup<TNegator, TOptimizations>(source0, source1, bitmap);
+                    if (result != Vector128<byte>.Zero)
+                    {
+                        return ComputeFirstIndex<TNegator>(ref searchSpace, ref currentSearchSpace, result);
+                    }
+
+                    currentSearchSpace = ref Unsafe.Add(ref currentSearchSpace, 2 * Vector128<short>.Count);
+                }
+                while (Unsafe.IsAddressLessThan(ref currentSearchSpace, ref twoVectorsAwayFromEnd));
+            }
+
+            // We have 1-16 characters remaining. Process the first and last vector in the search space.
+            // They may overlap, but we'll handle that in the index calculation if we do get a match.
+            {
```

Comment:
This logic appears to depend on there being at least a vector's worth of data.  I assume that's being checked by the caller, but it'd be good to assert it.